### ISSUE TITLE
Make organization slug and other necessary options position independent

### ISSUE THREAD

**Issue by john1jan** [getsentry/sentry-cli#179]
I am creating a release for my project using release new command

`sentry-cli releases new 1.5`

But getting this error

error: An organization slug is required (provide with --org)

Where can I set organization slug. If I use **--org** it 

**sentry-cli releases new 1.5 --org ORGANIZATION**
```
`error: Found argument '--org' which wasn't expected, or isn't valid in this context`

USAGE:
    sentry-cli releases new [OPTIONS] <VERSION>
```

**Comment by jan-auer** [comment]
Hi @john1jan, this is slightly confusing but you currently have to put the `--org` parameter at a specific position: 

```
sentry-cli releases --org <org> new --project <proj> <name>
```

Of course it should not matter where to put the options, so let's leave this open until we fix it.

**Comment by PatNeedham** [comment]
@jan-auer where does the `org` field appear on the Sentry dashboard for an existing project? When logged in and viewing a specific project at url https://sentry.io/$1/$2, does `$1` map to org slug and `$2` map to project slug?

**Comment by jan-auer** [comment]
@PatNeedham correct, the org and project slugs appear in the URL: `https://sentry.io/org-slug/project-slug/`. Besides that, we currently show the org slug in settings at the top.

**Comment by oliverbenns** [comment]
Sorry but this is pretty poor design. Now I'm trying all sorts of combinations to get the upload-sourcemaps script to work which does not follow the above ordering.
```
sentry-cli releases --org "org_name" files --project "proj_name" "release_name" upload-sourcemaps ./sourcemaps
```

**Comment by AlirezaNoorali** [comment]
I have the same problem. assume that my project name is **myapp** and my org name is **myorg** now how should I send the command? what is releases ?

**Comment by mir3z** [comment]
This is very unintuitive and documentation does not provie a single example of correct usage of these options in cli.

**Comment by tonyo** [comment]
As an alternative, there's an option of using environment variables, e.g.:
`SENTRY_ORG=org-name SENTRY_PROJECT=project-name sentry-cli releases new 1.0`
This might be more intuitive at the moment.

**Comment by XyrisKennEnrian** [comment]
This critical part of syntax example is not present in the documentation ("org_slug" & "proj_slug", and the project URL doesn't match what is described here. Left to guess at the author's intention.

**Comment by silavsale** [comment]
> @jan-auer & @PatNeedham & @john1jan :
> I have the same problem. assume that my project name is **myapp** and my org name is **myorg** and my release is **1.0**
> now, how should I send the command?
> it the bellow command right?
> `sentry-cli releases new 1.0 --org myorg new --project myapp`
> 
> I get this error:
> `error: Found argument '--org' which wasn't expected, or isn't valid in this context`

Did you found any solution how to set SENTRY_ORG?

**Comment by aarhusgregersen** [comment]
I know I'm chiming in a bit late on this, but the documentation has been updated to reflect this nicely. However, it may make sense to keep it open as the issue points to fixing option positions, which still isn't possible.

**Comment by indatawetrust** [comment]
This is the scariest cli app experience i have ever seen. I really haven't seen anything worse. Thanks sentry.

![gif](https://media.tenor.com/images/92ea8142d21b2894ea3ed62bbd81ef37/tenor.gif)

**Comment by PabloCarmona** [comment]
> I know I'm chiming in a bit late on this, but the documentation has been updated to reflect this nicely. However, it may make sense to keep it open as the issue points to fixing option positions, which still isn't possible.

Where is the URL to the exact place in the docs you are referring to? Thanks!

**Comment by kubijo** [comment]
As a possibly related topic, could you please change the `--project` flag to also accept project ID as opposed to just project slug?
This would very much simplify configuration of environments that need to use these commands (for example source maps upload) from needing 3 variables (dsn, token, project name) to 2 variables (dsn, token) because project ID is part of DSN

**Comment by kamilogorek** [comment]
Sure, I'm working on it right now. However, for source maps, you don't need a DSN to be set. We just need token and org+project.

**Comment by kubijo** [comment]
…and url. URL and project ID can be parsed from DSN, token is sadly needed as one extra variable

**Comment by kamilogorek** [comment]
url is only required if you are using on-premise instance. Unfortunately, we cannot change this, as the whole Sentry API is constructed around org-slug and project-slug, and CLI is fully reliant on it.

**Comment by kubijo** [comment]
Sure thing ... I'm just illustrating one use case where DSN can cover some attributes required by CLI, but project slug requirement imposes extra configuration burden… TLDR: I don't think we disagree on anything here :+1:

**Comment by kamilogorek** [comment]
Should be fixed in `1.74.1`.

Those will all work now:

```
sentry-cli releases new 1.0.0 --org acme --project app
sentry-cli releases new --org acme --project app 1.0.0
sentry-cli releases  --org acme --project app new 1.0.0
sentry-cli releases new 1.0.0 --org acme --project frontend --project backend
```

short version


```
sentry-cli releases new 1.0.0 -o acme -o app
sentry-cli releases new -o acme -p app 1.0.0
sentry-cli releases  -o acme -p app new 1.0.0
sentry-cli releases new 1.0.0 -o acme -p frontend -p backend
```
